include/RAJA/policy/sycl/scan.hpp

```diff
@@ -0,0 +1,236 @@
+/*!
+******************************************************************************
+*
+* \file
+*
+* \brief   Header file providing RAJA scan declarations.
+*
+******************************************************************************
+*/
+
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2016-24, Lawrence Livermore National Security, LLC
+// and RAJA project contributors. See the RAJA/LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+#ifndef RAJA_scan_sycl_HPP
+#define RAJA_scan_sycl_HPP
+
+#include <cstddef>
+#include "RAJA/config.hpp"
+#include "camp/resource/sycl.hpp"
+
+#if defined(RAJA_ENABLE_SYCL)
+
+#include "RAJA/policy/sycl/policy.hpp"
+
+namespace RAJA
+{
+namespace impl
+{
+namespace scan
+{
+
+template <size_t BLOCK_SIZE, bool Async, typename InputIter, typename Function>
+RAJA_INLINE 
+camp::resources::EventProxy<camp::resources::Sycl> 
+inclusive_inplace(
+    camp::resources::Sycl sycl_res,
+    ::RAJA::policy::sycl::sycl_exec<BLOCK_SIZE, Async>,
+    InputIter begin,
+    InputIter end,
+    Function binary_op)
+{
+    ::sycl::queue* sycl_queue = sycl_res.get_queue();
+ 
+    using valueT = typename std::remove_reference<decltype(*begin)>::type;
```

Comment:
Probably best to use iterator_traits https://en.cppreference.com/w/cpp/iterator/iterator_traits